include/SDL_pixels.h

```diff
@@ -627,7 +627,35 @@ extern DECLSPEC void SDLCALL SDL_GetRGBA(Uint32 pixel,
  */
 extern DECLSPEC void SDLCALL SDL_CalculateGammaRamp(float gamma, Uint16 * ramp);
 
+/**
+ * Read one pixel's worth of data for a given pixel format.
+ *
+ * \param data a pointer to the pixel's data
+ * \param format an SDL_PixelFormat structure describing the format of the pixel
+ * \returns a pixel value suitable for passing to SDL_GetRGB or SDL_GetRGBA
+ *
+ * \since This function is available since SDL 2.0.18
+ *
+ * \sa SDL_WritePixelValue
+ * \sa SDL_GetRGB
+ * \sa SDL_GetRGBA
+ */
+extern DECLSPEC Uint32 SDLCALL SDL_ReadPixelValue(void *data, SDL_PixelFormat *format);
```

Comment:
It might be better to make these macros, to avoid per-pixel function call overhead.

Reply:
I'm not quite sure how to make this into a macro without making it horrendously ugly, given it ideally wants to "return" a value. Unless we wanted to have it accept a pointer to a Uint32 in which to place the result?